test/celeritas/global/KernelContextException.test.cc

```diff
@@ -132,17 +134,27 @@ TEST_F(KernelContextExceptionTest, typical)
         EXPECT_EQ(SurfaceId{}, e.next_surface());
         if (CELERITAS_USE_JSON && !CELERITAS_USE_VECGEOM)
         {
-            EXPECT_EQ(
-                R"json({"dir":[0.0,0.0,1.0],"energy":[10.0,"MeV"],"event":1,"label":"test-kernel","num_steps":1,"particle":0,"pos":[0.0,1.0,5.0],"surface":11,"thread":15,"track":3,"track_slot":15,"volume":2})json",
-                get_json_str(e));
+            std::stringstream ss;
+            ss << R"json({"dir":[0.0,0.0,1.0],"energy":[10.0,"MeV"],"event":1,"label":"test-kernel","num_steps":1,"particle":0,"pos":[0.0,1.0,5.0],"surface":11,"thread":)json"
+               << e.thread().unchecked_get()
+               << R"json(,"track":3,"track_slot":15,"volume":2})json";
+            EXPECT_EQ(ss.str(), get_json_str(e));
         }
     };
     // Since tracks are initialized back to front, the thread ID must be toward
     // the end
+    Span track_slots{
+        step.core_data().states.track_slots[AllItems<TrackSlotId::size_type>{}]};
+    auto idx
+        = std::distance(track_slots.begin(),
+                        std::find(track_slots.begin(), track_slots.end(), 15));
+    EXPECT_NE(track_slots.size(), idx);
```

Comment:
Since you use the same function down below, I'd define a helper function:
```c++
ThreadId  find_thread(CoreRef<Memspace::host> const&, TrackSlotId) const;
```
you could also use this to restore the `ThreadId` check in the first test.

Reply:
@sethrj I would say `find_thread` is better as a free function since it doesn't need access to `KernelContextExceptionTest`